amazon/s3/client/src/main/kotlin/org/http4k/connect/amazon/s3/model/S3BucketPreSigner.kt

```diff
@@ -20,18 +22,39 @@ class S3BucketPreSigner(
     bucketName: BucketName,
     region: Region,
     credentialsProvider: CredentialsProvider,
-    clock: Clock = Clock.systemDefaultZone()
+    clock: Clock = Clock.systemDefaultZone(),
+    forcePathStyle: Boolean = false
 ) {
     constructor(
         bucketName: BucketName,
         region: Region,
         credentials: AwsCredentials,
-        clock: Clock = Clock.systemDefaultZone()
+        clock: Clock = Clock.systemDefaultZone(),
+        forcePathStyle: Boolean = false
     ) : this(
         bucketName = bucketName,
         region = region,
         credentialsProvider = { credentials },
-        clock = clock
+        clock = clock,
+        forcePathStyle = forcePathStyle
+    )
+
+    /**
+     * Convenience constructor to create an S3BucketPreSigner from an http4k Environment
+     */
+    constructor(
+        bucketName: BucketName,
+        region: Region,
+        env: Environment = Environment.ENV,
+        clock: Clock = Clock.systemDefaultZone(),
+        forcePathStyle: Boolean = false,
+        credentialsProvider: CredentialsProvider = CredentialsProvider.Environment(env)
```

Comment:
This uses a similar signature as the `S3Bucket.Http` function.
And I would actually also use a default `region = AWS_REGION(env)` (requires a reordering of the parameters).
I haven't done that since it's not there for `bucketRegion` in `S3Bucket.Http`. Is there intentionally no default for `bucketRegion`?

Reply:
The `S3BucketPresigner` primary constructor mirrors the `S3Bucket` primary constructor.  The lack of a secondary constructor taking an `Envionment` was an oversight.

I'm not sure if there was any reason for the lack of the default `Region` in the matching `S3Bucket` constructor.

Reply:
Yeah - you're right - the map env is pointless and just adding noise. I'll deprecate them all now, bump the second version digit and we can move to just using the Environnent version.

Reply:
On second thought, I'll leave them undeprecated, but flip the defaults to using the Environment version. 🙃